tools/azsdk-cli/Azure.Sdk.Tools.Cli/Tools/GitHubLabelsTool/GitHubLabelsTool.cs

```diff
@@ -0,0 +1,228 @@
+                    return $"Service label '{label}' already exists.";
+                }
+                else if (checkResult == LabelHelper.ServiceLabelStatus.NotAServiceLabel)
+                {
+                    logger.LogWarning($"Label '{label}' exists but is not a service label. No action taken.");
+                    return $"Label '{label}' exists but is not a service label. Try a different label.";
+                }
+
+                var branchResult = await githubService.CreateBranchAsync("Azure", "azure-sdk-tools", $"add_service_label_{normalizedLabel}", "main");
+                logger.LogInformation($"Branch creation result: {branchResult}");
+
+                // If branch already exists, return early with the compare URL
+                if (branchResult.Contains("already exists"))
+                {
+                    var branchResponse = new GenericResponse()
+                    {
+                        Details = {
+                            $"Result: {branchResult}"
+                        }
+                    };
+                    return output.Format(branchResponse);
+                }
+
+                logger.LogInformation($"Creating new service label: {label}. Documentation link: {link}");
+
+                // Update the common-labels.csv file
+                var csvContent = await githubService.GetContentsAsync("Azure", "azure-sdk-tools", "tools/github/data/common-labels.csv");
+
+                if (csvContent == null || csvContent.Count == 0)
+                {
+                    throw new InvalidOperationException("Could not retrieve common-labels.csv file");
+                }
+
+                var csvContentString = csvContent[0].Content;
+
+                var updatedFile = labelHelper.CreateServiceLabel(csvContentString, label); // Contains updated CSV content with the new service label added
+
+                await githubService.UpdateFileAsync("Azure", "azure-sdk-tools", "tools/github/data/common-labels.csv", $"Adding {label}", updatedFile, csvContent.First().Sha, $"add_service_label_{normalizedLabel}");
+
+                // Create the pull request
+                var result = await githubService.CreatePullRequestAsync(
+                    repoName: "azure-sdk-tools",
+                    repoOwner: "Azure",
+                    baseBranch: "main",
+                    headBranch: $"add_service_label_{normalizedLabel}",
+                    title: $"Add service label: {label}",
+                    body: $"This PR adds the service label '{label}' to the repository. Documentation link: {link}",
+
+                    // TODO: Before merge, make this not draft
+                    draft: true
+                );
+
+                logger.LogInformation($"Service label '{label}' pull request created successfully. Result: {string.Join(", ", result)}");
+
+                var response = new GenericResponse()
```

Comment:
We should look into whether we should use `GenericResponse` ... if so, use just about everywhere, not jus there. If not, probably just string output.